### PR TITLE
BUG: Adding wrap around checking in WCS

### PR DESCRIPTION
**Pull request by ceb8**
### Description

This pull request adds checking to the linear fit function to handle WSCs near the 360->0 degree crossover. In this case your can get incorrectly large residuals between eg. 359.8 deg and 0.08 deg. This PR fixes that issue.

@shannnnnyyy @eteq  should review this.

### CODE REVIEW

**Comment by bsipocz** [comment]
@ceb8 - based on the description I felt like this is a bugfix, if you agree please move the changelog to 4.0.2. However if I misunderstood, then just let me know and I'll remilestone.

**Comment by bsipocz** [comment]
(Also, Clare needs to be added to the github team to be able to assign her to review, and I don't have a power to do that).

File: astropy/wcs/utils.py [hunk 1]
```diff
@@ -890,7 +890,10 @@ def _linear_wcs_fit(params, lon, lat, x, y, w_obj):
     w_obj.wcs.cd = ((cd[0], cd[1]), (cd[2], cd[3]))
     w_obj.wcs.crpix = crpix
     lon2, lat2 = w_obj.wcs_pix2world(x, y, 0)
+
     resids = np.concatenate((lon-lon2, lat-lat2))
+    resids[resids > 180] = 360 - resids[resids > 180]
+    resids[resids < -180] = 360	+ resids[resids < -180]
```
Comment (pllim) [inline astropy/wcs/utils.py:896]: Sneaky tab.
```suggestion
    resids[resids < -180] = 360 + resids[resids < -180]
```

Comment (ceb8) [reply astropy/wcs/utils.py:896]: How did that get in there???? 😉

**Comment by ceb8** [comment]
@bsipocz Yes this is definitely a bugfix. It was supposed to go into the original PR, but must have got lost in the shuffle.

**Comment by ceb8** [comment]
@pllim I'll work on a test.

**Comment by pllim** [comment]
> It was supposed to go into the original PR

@ceb8 , can you please cross-link that PR here for future reference? Thanks!

**Comment by shannnnnyyy** [comment]
I am investigating a coordinate indexing issue that @ceb8 brought to my attention. The fitting works, but there may be some ambiguity of 0 vs 1 indexing that is confusing. I'm poking at it to try to figure out what's going on, and hopefully it will just require a clarification in the docs of what the expected input should be.

A test that covers a case where the 0>360 crossover should be added. I think the existing test can be used with a new header-string that describes this case? Otherwise I have no other comments!

**Comment by ceb8** [comment]
@pllim  https://github.com/astropy/astropy/pull/7884 was the first one, which got replaced by https://github.com/astropy/astropy/pull/9469.
Pretty sure what happened is the 0->360 checking got added to the wcs with sip fitting function which made everyone think it was done, so it never got added to the linear wcs fitting function.

**Comment by ceb8** [comment]
Added a test and confirmed that it fails without bugfix, but passes with bugfix.

**Comment by ceb8** [comment]
@shannnnnyyy After further investigation I think the indexing issue was a red herring and actually everything is fine on that front. As least as far as I can test.

So I think all this PR needs is a review from an approved reviewer.

**Comment by nden** [comment]
Thanks @ceb8

**Pull request merged**